code/game/objects/effects/effect_system.dm

```diff
@@ -839,11 +839,9 @@ steam.start() -- spawns the effect
 			reagents.reaction(M)
 		return
 
-	if (istype(AM, /mob/living/carbon))
+	if(istype(AM, /mob/living/carbon))
 		var/mob/living/carbon/M = AM
-		if (M.Slip(5, 2, 1))
-			to_chat(M, "<span class='notice'>You slipped on the foam!</span>")
-
+		M.Slip(5, 2, 1, slipped_on = src, onwhat = "the foam")
```

Comment:
why do you need an onwhat arg? you're already passing src, so it can use the name, right?

Reply:
For situations like slipping on wet/lubed/icy floors, and clown PDAs.
Current behavior for floors is:
wet floor -> "the wet floor!"
lubed floor -> "the floor!"
icy floor -> "the icy floor!"
I wanted to preserve that as well as make it as flexible as possible for any other cases where the `name` of the thing being slipped on isn't what is displayed in the slip message.
In this particular case fire suppression foam also exists but it still says "the foam!" when slipped upon.